### PR TITLE
feat(checkbox): improve disabled state and create disable mixin

### PR DESCRIPTION
**Pull request by BenOsodrac**
Issue number: internal

---------

## What is the new behavior?

- Improved Checkbox disabled state
- Added more examples on theme-ionic file for disabled checkbox
- Created ionic.mixin for new DS specific mixins, while still forwarding the globals one.
- Created mixin for the disabled-state, that can be reused on multiple Ionic Theme components.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/themes/ionic/ionic.mixins.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@use "../../foundations/ionic.vars.scss" as tokens;
+
+// Disabled mixin to be used on pseudo elements for the Ionic Theme disabled state
+//
+// ex: :host(.checkbox-disabled) .native-wrapper:after {
+//       @include globals.disable-state();
+//      }
+//
+// --------------------------------------------------
+@mixin disable-state() {
```
Comment (brandyscarney) [inline core/src/themes/ionic/ionic.mixins.scss:15]:
```suggestion
@mixin disabled-state() {
```

File: core/src/components/checkbox/checkbox.ionic.scss [hunk 2]
```diff
@@ -319,11 +319,8 @@ input {
   pointer-events: none;
 }
 
-// disabled, indeterminate checkbox
-:host(.checkbox-disabled.checkbox-indeterminate) .checkbox-icon {
-  border-width: globals.$ionic-border-size-0;
-
-  background-color: #{globals.$ionic-color-base-white};
+:host(.checkbox-disabled) .native-wrapper:after {
+  @include globals.disable-state();
```
Comment (brandyscarney) [inline core/src/components/checkbox/checkbox.ionic.scss:323]:
```suggestion
  @include globals.disabled-state();
```

**Review by brandyscarney — APPROVED** [review]
Minor name change requested to the mixin, but other than that this looks good! I like the idea of moving it to a mixin. 👍

**Review by gnbm — APPROVED** [review]
LGTM 🚀

**Pull request merged**